### PR TITLE
[ACR] Change to support python3.6 or above only

### PR DESCRIPTION
**Pull request by YalinLi0312**

### CODE REVIEW

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def extract_data(pipeline_response):
             link = None
             if "Link" in pipeline_response.http_response.headers.keys():
                 link = _parse_next_link(pipeline_response.http_response.headers["Link"])
-            elif "link" in pipeline_response.http_response.headers.keys():  # python 2.7 turns this into lowercase
+            elif "link" in pipeline_response.http_response.headers.keys():
```
Comment (lmazuel) [inline sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py:186]: We shall either remove the if, or leave a comment "This test was for Py2 only and we should consider to remove it". IF you just remove the comment, in 6 months we will not remember why this test is here at all :)

File: sdk/containerregistry/azure-containerregistry/README.md [hunk 2]
```diff
@@ -23,7 +27,8 @@ pip install --pre azure-containerregistry
 
 ### Prerequisites
 
-You need an [Azure subscription][azure_sub] and a [Container Registry account][container_registry_docs] to use this package.
+* Python 3.7 or later is required to use this package.
```
Comment (lmazuel) [inline sdk/containerregistry/azure-containerregistry/README.md:30]: 3.6

File: sdk/containerregistry/azure-containerregistry/dev_requirements.txt [hunk 3]
```diff
@@ -2,6 +2,6 @@
 -e ../../../tools/azure-devtools
 ../../core/azure-core
 ../azure-mgmt-containerregistry
-aiohttp>=3.0; python_version >= '3.5'
+aiohttp>=3.0; python_version >= '3.7'
```
Comment (lmazuel) [inline sdk/containerregistry/azure-containerregistry/dev_requirements.txt:5]: 3.6

File: sdk/containerregistry/azure-containerregistry/samples/README.md [hunk 4]
```diff
@@ -23,7 +23,7 @@ The async versions of the samples require Python 3.6 or later.
 |[sample_list_tags.py][list_tags] and [sample_list_tags_async.py][list_tags_async] | List tags on an image using an anonymous access |
 
 ### Prerequisites
-* Python 2.7, or 3.6 or later is required to use this package.
+* Python 3.7 or later is required to use this package.
```
Comment (lmazuel) [inline sdk/containerregistry/azure-containerregistry/samples/README.md:26]: 3.6

File: sdk/containerregistry/azure-containerregistry/setup.py [hunk 5]
```diff
@@ -43,13 +37,12 @@
     classifiers=[
         "Development Status :: 4 - Beta",
         "Programming Language :: Python",
-        "Programming Language :: Python :: 2",
-        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3 :: Only",
         "Programming Language :: Python :: 3",
-        "Programming Language :: Python :: 3.6",
```
Comment (lmazuel) [inline sdk/containerregistry/azure-containerregistry/setup.py:49]: Keep 3.6

File: sdk/containerregistry/azure-containerregistry/setup.py [hunk 6]
```diff
@@ -59,13 +52,11 @@
             "azure",
         ]
     ),
+    python_requires=">=3.7",
```
Comment (lmazuel) [inline sdk/containerregistry/azure-containerregistry/setup.py:55]: 3.6

File: sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py (hunk 1 above)
Comment (YalinLi0312) [reply sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py]: Removed the elif section for python 2.7

**Review by lmazuel — APPROVED** [review]

**Pull request merged**